File: Problem 012/Solution 12.c
/* Solves http://projecteuler.net/problem=12
 * Requires -lm linker flags on some compilers.
 * gcc Solution\ 12.c -lm */

#include <stdio.h>
#include <math.h>

int main()
{
	/* vars */
	int number = 0;
	int check;
	int counter = 1;
	int divisors;

	/* Loop while divisors is under 500 */
	while(divisors <= 250)
	{
		number += counter;
		check = 1;
		divisors = 0;

		/* Check every number under number/2 */
		for(check = 1; check <= sqrt(number); ++check)
			if(number % check == 0)
				++divisors;

		++counter;

	}

	printf("Answer: %d\r\n", number);

	return 0;
}

File: Problem 012/Solution 12.rb
# Solves http://projecteuler.net/problem=12

number = 0
check = 0
counter = 1
divisors = 0

while (divisors <= 250)
	puts number
	number += counter
	check = 1
	divisors = 0

	for check in (1..Math.sqrt(number))
		divisors += 1 if (number % check == 0)
	end

	counter += 1
end

print "Answer: ", number, "\r\n"
